### ISSUE TITLE
Add support for icons in Button blocks

### ISSUE THREAD

**Issue by truchot** [WordPress/gutenberg#16513]
In the same idea of #16480, #16481 or #16482 (improvements of button block), maybe we could think about adding a suffix/prefix icon to the button label ?

**Comment by mtias** [comment]
Agree this would be cool. We probably want to see where https://github.com/WordPress/gutenberg/issues/16484 goes first.

**Comment by deborah86** [comment]
I am surprised this feature has not been added yet and it hasn't gotten any traction.

**Comment by Julianoe** [comment]
Has there been some new discussions about icons in Button block recently? The "button with icon" component is very very very common in UIs and has been for years. Currently there is no simple way for users to do that without installing a plugin. Why?

**Comment by Jabe64** [comment]
> Has there been some new discussions about icons in Button block recently? The "button with icon" component is very very very common in UIs and has been for years. Currently there is no simple way for users to do that without installing a plugin. Why?

As @mtias mentioned #16484 goes first but it's true there hasn't been any update about it for a while 😔

**Comment by Julianoe** [comment]
Problem is that there is currently no way to do this in other way than inline image, a feature that seems to have been completely forgotten because it's all broken. 

Apart from the very obvious bugs (see my just posted issue #45664), the inline image does not behave like an image block that would be contained inside the button which is, I think, the expected behavior. That prevents for example to use .svg images on a WP install where they are reauthorized. 

Another way would be to have custom html block inside a button block to include the svg code of an icon.. which is also impossible.

However the Icon Block situation happens to be resolved, I wonder if the two use cases could not be easier ways to implement a "icon button" in the current state of things.

**Comment by sathwikprabhu** [comment]
I use this as of now: https://github.com/ndiego/enable-button-icons

**Comment by Jiwoon-Kim** [comment]
A short follow-up, with one correction.

I said earlier that the Button block renders an `<a>` rather than a semantic `<button>`. That is out of date and I should have checked before writing it: `tagName` has accepted `button` since PR #54206, and the HTML element control has been in the Advanced panel since PR #70139. The semantics objection I raised does not apply — one icon slot here would serve both renderings, which makes the case for adding it stronger rather than weaker.

The other half of my comment was about the icon API being SVG-based and constraining themes. I still think there is something to settle there, but it is not really a `core/button` question, so I have opened it separately rather than continue it here:

**#82700** — Define a state-aware icon reference contract for blocks

The short version: whether an `icon` attribute needs one reference or two is decided by the icon source rather than by the block, because a registered icon is one SVG with no variant axis. A single-reference attribute works for a variable icon font and renders nothing for the second state on the registry path. `core/accordion-heading` already runs into this and solves it by rotating a literal `+` 45°, which works only for that one glyph.

That issue includes a working reference implementation of the markup contract, so if icons do land here, there is something concrete to argue against.

**Comment by landwire** [comment]
Hi there,
when you consider this please make sure you allow also for a button to be **only** an icon and the label can be empty or only shows for screen reader users, as sometimes you just want an icon button and no label on it.

**Comment by Jiwoon-Kim** [comment]
@landwire 
I made a small reference implementation for exactly this case:

https://jiwoon-kim.github.io/axismundi/styleguide/components/icon-buttons/

It treats an icon-only button as a distinct component from a text Button with a leading/trailing icon. The visible content is only the icon, while the accessible name remains in ".screen-reader-text":

```
<button class="wp-block-axismundi-icon-button">
	<span class="material-symbols-outlined" aria-hidden="true">add</span>
	<span class="screen-reader-text">Add</span>
</button>
```

The demo also covers toggle/non-toggle states, sizing, width variants, and target size.

If "core/icon-button" is ever considered separately from adding icons to "core/button", this could be a useful markup reference.
